javascript/core/array-methods/how-to-merge-two-arrays.md

```diff
@@ -33,24 +18,44 @@ links:
 ---
 ## Content
 
-`concat` is not always the best approach to merging arrays as it generates a new array.
+One approach that we can use to merge two arrays in JavaScript, is to loop through the first array and add each of its elements to the second array:
 
-Since `var`s are passed by reference, `concat` may mess up a reference.
+```js
+let array1 = [1, 2, 3];
+let array2 = [4, 5, 6];
 
-Instead:
+for (let item1 of array1) {
+  array2.push(item1);
+}
 
-```javascript
-var array1 = [1, 2, 3];
-var array2 = [4, 5, 6];
+console.log(array2);
+// 1, 2, 3, 4, 5, 6
 ```
 
-Merge them, adding the second one after the first one:
+We can also use the spread operator, i.e. `...`, to do the same with less code:
 
-```javascript
-Array.prototype.push.apply(array1, array2);
+```js
+let array1 = [1, 2, 3];
+let array2 = [4, 5, 6];
+
+array2.push(...array1);
+
+console.log(array2);
+// 1, 2, 3, 4, 5, 6
+```
+
+Arrays also have a method that can be used to merge them:
+
+```js
+let array1 = [1, 2, 3];
+let array2 = [4, 5, 6];
+let array3 = array1.concat(array2);
 ```
 
-Unlike other languages, JavaScript does not support `+` for concatenating arrays. The output of such an operation would be a string:
+The notable difference when using `concat` is that it, instead of merging one array into another, it generates a new array out of the two merged arrays.
```

Comment:
```suggestion
The notable difference when using `concat` is that, instead of merging one array into another, it generates a new array out of the two merged arrays.
```